.github/workflows/test-pr.yml

```diff
@@ -30,6 +30,30 @@ jobs:
             git push origin HEAD:${GITHUB_HEAD_REF}
           fi
 
+  arch:
+    name: 'Arch Linux Package'
+    runs-on: [self-hosted, linux, normal]
+    steps:
+      - uses: actions/checkout@v3
+      - name: 'Build and Test Package'
+        uses: ./.github/actions/test-package
+        with:
+          dockerfile: package/arch/Dockerfile
+          os: archlinux
+          distro: base
+          build-package: package/arch/build-package
+          test-package: package/arch/test-package
+          pkg-name: kframework_arch_x86_64.pkg.tar.zst
+      - name: 'Upload Package to Release'
+        uses: actions/upload-release-asset@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.JENKINS_GITHUB_PAT }}
+        with:
+          asset_content_type: 'application/gzip'
+          asset_name: 'K Framework Arch Package'
+          asset_path: kframework_arch_x86_64.pkg.tar.zst
```

Comment:
And here we   do the asset upload (so this needs to match the prior step).

Reply:
Hmm interesting my first thought is to check the cwd during the execution and maybe that's not what we expected. 
My brain is a bit mush at the moment, I can come back tomorrow morning and give it a run through manually to see if something jumps out at me.